sdk/python/packages/flet-core/src/flet_core/flashlight.py

```diff
@@ -0,0 +1,62 @@
+from typing import Any, Optional
+
+from flet_core.control import Control
+from flet_core.ref import Ref
+
+
+class Flashlight(Control):
+    """
+    A control to use FlashLight. Works on iOS and Android. Based on torch_light Flutter widget (https://pub.dev/packages/torch_light).
+
+    Flashlight control is non-visual and should be added to `page.overlay` list.
+
+    Example:
+    ```
+    import flet as ft
+
+    def main(page: ft.Page):
+        flashLight = ft.Flashlight()
+        page.overlay.append(flashLight)
+        page.add(
+            ft.TextButton("toggle", on_click: lambda _: flashlight.toggle())
+        )
+
+    ft.app(target=main)
+    ```
+
+    """
+
+    def __init__(
+        self,
+        ref: Optional[Ref] = None,
+        data: Any = None,
+    ):
+        Control.__init__(
+            self,
+            ref=ref,
+            data=data,
+        )
+
+        self.turned_on = False
+
+    def _get_control_name(self):
+        return "flashlight"
+
+    def turn_on(self, wait_timeout: Optional[int] = 3) -> bool:
+        sr = self.invoke_method("on", wait_for_result=True, wait_timeout=wait_timeout)
+
+        if int(sr) == 1:
+            self.turned_on = True
+        return self.turned_on
+
+    def turn_off(self, wait_timeout: Optional[int] = 3) -> bool:
+        sr = self.invoke_method("off", wait_for_result=True, wait_timeout=wait_timeout)
```

Comment:
I think we need `_async` counterparts for these methods as well. When there is a method waiting for the result it should have `_async` friend as well. Here you are suppressing the issue by giving timeout.

Reply:
yep